tests/test_sensor.py

```diff
@@ -80,6 +86,82 @@ def status_func(value):
     assert sensor.status == Sensor.Status.NOMINAL
 
 
+class MyEnum(enum.Enum):
+    ZERO = 0
+    ONE = 1
+
+
+class OtherEnum(enum.Enum):
+    ABC = 0
+    DEF = 1
+
+
+@pytest.mark.parametrize(
+    "sensor_name,sensor_type,default,compatible_value",
+    [
+        ("bool", bool, False, True),
+        ("int", int, 0, np.int32(1234)),
+        ("float", float, 0.0, np.int32(1234678)),
+        ("float", float, 0.0, np.float32(12345678)),
+        ("str", str, "zero", "one-two-three-four"),
+        ("bytes", bytes, b"zero", b"one-two-three-four"),
+        ("address", Address, Address(IPv4Address("0.0.0.0")), Address(IPv4Address("1.2.3.4"))),
+        ("timestamp", Timestamp, Timestamp(0.0), 12345678),
+        ("enum", MyEnum, MyEnum.ZERO, MyEnum.ONE),
+    ],
+)
+def test_sensor_value_setter_success(
+    sensor_name: str,
+    sensor_type: Type[_T],
+    default: Any,
```

Comment:
I'd updated this to be `default: _T`, but `mypy` complained it should be `"str"`. I'm not sure if it's parsing the parameterised list before reporting this.

Either way, let me know if I should update/debug/`#type: ignore`.

Reply:
That's mypy telling you that you've made a mistake: you're passing `default` as the sensor description.

But I think you might have misunderstood my comments about defaults. My intention was not that you pass in an explicit default that is just the same thing aiokatcp would have used as the default anyway. I think this test should have two parts:

1. Create a sensor with no explicit default, assign the `compatible_value`, then check that the value is correct.
2. Create a second sensor where `compatible_value` is set as the default, then check that the value is correct.

Note that the name `default` is a little misleading: a better name would have been `initial_value`.